### PR TITLE
Automatic Pastebin Parsing

### PR DESCRIPTION
**Pull request by underscore11code**
Parses pastebin links from supported sites for common issues, and informs the user if it finds any, with info on how to remedy them.
Currently supports:
- [bytebin](https://bytebin.lucko.me)
- [hasteb.in](https://hasteb.in)
- [hastebin](https://hastebin.com)
- [paste.helpch.at](https://paste.helpch.at)
- [paste.lucko.me](https://paste.lucko.me)
- [Github Gists](https://gist.github.com)
- [Pastebin](https://pastebin.com)

As well as all errors on the [Storage Errors](https://github.com/lucko/LuckPerms/wiki/Storage-system-errors) page. If you have other pastebin or error suggestions, let me know.

Also, this is my first Javascript PR, feel free to yell at me about code style/practices, and I'll fix.

### CODE REVIEW

**Comment by lucko** [comment]
This looks great! :tada:

I have an additional challenge if you (or anyone else) are up for it! 

Discord supports sending files directly through the chat window, like so:



It would be awesome if clippy could grab the file (if it looks like a log/config file - maybe if it has a `.log` or `.yml` extension?), upload it to bytebin, and then send the link to the channel.

And additionally do all of the same parsing logic as done by this PR ;)

If nobody is keen to take this on now - that's cool - we can get this merged and then I'll tackle the extra stuff another time :)

**Comment by Turbotailz** [comment]
Yesss I was planning on adding that functionality myself but if anyone else wants to give it a try then that's cool. Should be pretty easy to do.

**Comment by Turbotailz** [comment]
Actually, just thought of a really good way to let users share their configs with us without them worrying about database details being sharing - they can upload the config.yml which Clippy then deletes immediately and parses the file to check for any errors.

File: modules/checks.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    {checks: [/Caused by: com\.mysql\.jdbc\.exceptions\.jdbc4\.MySQLSyntaxErrorException: User '\w+' has exceeded the 'max_user_connections' resource \(current value: \w+\)/,],
+    title: 'MySQL exceeded max connections',
+    link: 'https://github.com/lucko/LuckPerms/wiki/Storage-system-errors#mysql-exceeded-max-connections'},
+]
+module.exports = function (client) {
+    client.on('message', async message => {
+        if (message.channel.type !== 'text') return;
+        let getLink = '';
+        let originalLink = '';
+        checks.every(function(element, _){
```
Comment (Turbotailz) [inline modules/checks.js:38]: What's this `_` argument? It's not used in the callback. Also this callback can be an arrow statement.

File: modules/autoupload.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+const contentTypes = ['application/json', 'text/plain', 'text/yaml'];
+const bytebin = 'https://bytebin.lucko.me';
+
+module.exports = client => {
+    client.on('message', async message => {
+        if (message.channel.type !== 'text' || message.author.bot) return;
+        if (!message.attachments) return;
+        for (let attachment of message.attachments.values()) {
+            if (!contentTypes.some(type => mimeType.lookup(attachment.url) === type)) continue;
+            let content = await axios.get(attachment.url);
```
Comment (Turbotailz) [inline modules/autoupload.js:13]: We should be handling any errors here with the HTTP requests

File: modules/autoupload.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+const axios = require('axios');
+
+const contentTypes = ['application/json', 'text/plain', 'text/yaml'];
+const bytebin = 'https://bytebin.lucko.me';
+
+module.exports = client => {
+    client.on('message', async message => {
+        if (message.channel.type !== 'text' || message.author.bot) return;
+        if (!message.attachments) return;
+        for (let attachment of message.attachments.values()) {
```
Comment (Turbotailz) [inline modules/autoupload.js:11]: You can use `const` here as you are not reassigning `attachment`

File: modules/checks.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
+const axios = require('axios');
+const RichEmbed = require('discord.js').RichEmbed;
+const checks = [
```
Comment (Turbotailz) [inline modules/checks.js:3]: With the objects you're defining here, It's much better if you use the following syntax...

For single line objects (notice the spaces inside the brackets):
```js
{ regex: /https?:\/\/hastebin\.com\/(\w+)(?:\.\w+)?/g, getLink: 'https://hastebin.com/raw/{code}' }
```

And for multi-line objects:
```js 
{
    checks: [/Establishing SSL connection without server's identity verification is not recommended\./],
    title: 'MySQL SSL Error',
    link: 'https://github\.com/lucko/LuckPerms/wiki/Storage-system-errors#mysql-ssl-errors'
}
```

Just makes it easier to read 👌

File: modules/checks.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        if (message.channel.type !== 'text') return;
+        let getLink = '';
+        let originalLink = '';
+        checks.every(function(element, _){
+            let match = element.regex.exec(message.content);
+            if (match) {
+                getLink = element.getLink.replace('{code}', match[1]);
+                originalLink = match[0];
+                return false;
+            }else{
```
Comment (Turbotailz) [inline modules/checks.js:44]: Spaces around the brackets - `} else {`

File: modules/checks.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                return true;
+            }
+        })
+        if (!getLink) return;
+        let response = '';
+        try {
+            console.log(`Getting pastebin ${getLink}`)
```
Comment (Turbotailz) [inline modules/checks.js:51]: Don't forget semi-colons ;)

**Comment by Turbotailz** [comment]
Can we perhaps move the regex tests to a json file instead of putting them in the code?

**Comment by lucko** [comment]
Are the comments above resolved? Good to merge?!

**Comment by underscore11code** [comment]
I've fixed all the above, but someone should probably do a final check to make sure I don't have some other style thing derped.

**Comment by lucko** [comment]
Looks good to me!

**Pull request merged**